include/sys/arch_interface.h

```diff
@@ -216,6 +216,14 @@ typedef FUNC_NORETURN void (*arch_cpustart_t)(void *data);
  */
 void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 		    arch_cpustart_t fn, void *arg);
+
+/**
+ * @brief Return CPU power status
+ *
+ * @param cpu_num Integer number of the CPU
+ */
+bool arch_cpu_is_powered_up(int cpu_num);
```

Comment:
Would suggest something like "arch_cpu_active()".  Even on CAVS, there is a distinction between "powered up" (a bit set in the ADSPCS host register) and "active" (i.e. running Zephyr code and not in ROM or OS initialization, which is what you surely want to know).

Reply:
ack